addons/mail/static/src/new/core/thread_model.js

```diff
@@ -27,72 +50,66 @@ export class Thread {
      * @returns {Thread}
      */
     static insert(state, data) {
-        let thread;
         if (data.id in state.threads) {
-            thread = state.threads[data.id];
-        } else {
-            thread = new Thread();
-            thread._state = state;
+            const thread = state.threads[data.id];
+            thread.update(data);
+            return thread;
         }
-        thread.update(data);
-        state.threads[thread.id] = thread;
+        const thread = new Thread(state, data);
```

Comment:
this no longer assign thread into state in the create case

Reply:
actually it does but in constructor, this is a bit confusing to split the 2 state lines in different places